src/Common/src/System/Net/SocketAddressPal.Unix.cs

```diff
@@ -10,156 +10,129 @@ namespace System.Net
+
+                default:
+                    Debug.Fail("Unexpected failure in GetAddressFamily");
+                    throw new PlatformNotSupportedException();
             }
         }
 
+        public static unsafe AddressFamily GetAddressFamily(byte[] buffer)
+        {
+            AddressFamily family;
+            Interop.Error err;
+            fixed (byte* rawAddress = buffer)
+            {
+                err = Interop.Sys.GetAddressFamily(rawAddress, buffer.Length, (int*)&family);
+            }
+
+            ThrowOnFailure(err);
+            return family;
+        }
+
         public static unsafe void SetAddressFamily(byte[] buffer, AddressFamily family)
         {
+            Interop.Error err;
             fixed (byte* rawAddress = buffer)
             {
-                var sockaddr = (Interop.libc.sockaddr*)rawAddress;
-                Interop.CheckBounds((byte*)sockaddr, buffer.Length, &sockaddr->sa_family);
-
-                switch (family)
-                {
-                    case AddressFamily.Unspecified:
-                        sockaddr->sa_family = Interop.libc.AF_UNSPEC;
-                        break;
-
-                    case AddressFamily.Unix:
-                        sockaddr->sa_family = Interop.libc.AF_UNIX;
-                        break;
-
-                    case AddressFamily.InterNetwork:
-                        sockaddr->sa_family = Interop.libc.AF_INET;
-                        break;
-
-                    case AddressFamily.InterNetworkV6:
-                        sockaddr->sa_family = Interop.libc.AF_INET6;
-                        break;
-
-                    default:
-                        Debug.Fail("Unsupported addres family");
-                        throw new PlatformNotSupportedException();
-                }
+                err = Interop.Sys.SetAddressFamily(rawAddress, buffer.Length, (int)family);
             }
+
+            ThrowOnFailure(err);
         }
 
         public static unsafe ushort GetPort(byte[] buffer)
         {
             ushort port;
-
+            Interop.Error err;
             fixed (byte* rawAddress = buffer)
             {
-                var sockaddr = (Interop.libc.sockaddr*)rawAddress;
-                switch (sockaddr->sa_family)
-                {
-                    case Interop.libc.AF_INET:
-                        port = Interop.CheckedRead((byte*)sockaddr, buffer.Length, &((Interop.libc.sockaddr_in*)sockaddr)->sin_port);
-                        break;
-
-                    case Interop.libc.AF_INET6:
-                        port = Interop.CheckedRead((byte*)sockaddr, buffer.Length, &((Interop.libc.sockaddr_in6*)sockaddr)->sin6_port);
-                        break;
-
-                    default:
-                        Debug.Fail("Unsupported address family");
-                        throw new PlatformNotSupportedException();
-                }
+                err = Interop.Sys.GetPort(rawAddress, buffer.Length, &port);
```

Comment:
I don't have a strong preference either way.  For something like a ```byte[]```, where there's a clear mapping between it and a ```byte*```, I have no problem with ```byte[]``` being used in the P/Invoke signature, as it's really just a managed representation of the native concept; if it's the same performance to use a ```byte[]``` as it is to manually use fixed and pass a ```byte*```, then using a ```byte[]``` would have no real downside and an upside of less explicitly unsafe code and less code to be read and maintained.  But if there are other advantages, like maybe this is faster for some reason, or maybe we have other call sites that would benefit more from it being a ```byte*```, then it's fine.